magma/lib/magma/attributes/foreign_key.rb

```diff
@@ -11,7 +11,9 @@ def entry(value, loader)
       return [ foreign_id, nil ] if value.nil?
 
       if value.is_a? Magma::TempId
-        [ foreign_id, value.real_id ]
+        id = value.real_id
+        raise Magma::LoadFailed(["#{model_name}.#{attribute_name} value was not a valid #{link_model.model_name} identifier."]) if id.nil?
```

Comment:
the `real_id` method can return nil, as which happens when the record_entry itself is nil.  https://rollbar.com/ucsf/magma/items/661/